src/rpc/blockchain.cpp

```diff
@@ -28,12 +28,13 @@ using namespace std;
 extern void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry);
 void ScriptPubKeyToJSON(const CScript& scriptPubKey, UniValue& out, bool fIncludeHex);
 
-/* Calculate the difficulty for a given block index,
- * or the block index of the given chain.
+/* Calculate the difficulty for a given block index.
  */
-double GetDifficultyINTERNAL(const CChain& chain, const CBlockIndex* blockindex, bool networkDifficulty)
+double GetDifficultyINTERNAL(const CBlockIndex* blockindex, bool networkDifficulty)
 {
-    assert(blockindex);
+    if (blockindex == nullptr) {
+        return 1.0;
+    }
```

Comment:
Two commits were cherry-picked in the wrong order, leading to inconsistent results.

The order as of this review:
- 3e1ee310437f4c93113f6121425beffdc94702c2
  - Replaces `chainTip` with a `chain` argument.
- 343b98cbcd52854ffffaca7ada72f5522117ac7f
  - Replaces the entire chain-or-blockindex block with `assert(blockindex)`.
- ebec7317ca1acbc65afa7fb08fc219c315fc4527
  - Reverts the `assert(blockindex)` to the `blockindex == nullptr` check.

The order should be:
- 3e1ee310437f4c93113f6121425beffdc94702c2
  - Replaces `chainTip` with a `chain` argument.
- ebec7317ca1acbc65afa7fb08fc219c315fc4527
  - Reduces the entire chain-or-blockindex block to the `blockindex == nullptr` check.
- 343b98cbcd52854ffffaca7ada72f5522117ac7f
  - Replaces the the `blockindex == nullptr` check with `assert(blockindex)`.